### PR TITLE
Members and projects

### PR DESCRIPTION
**Pull request by ahnl**
Members and projects related additions to API routes and database

This the second attempt for #9

### CODE REVIEW

**Comment by Esinko** [comment]
Before this branch can be merged, we need to define what License this project uses. The utility script `discord.sh` requires us to use GPLv3 which in my opinion is not a favorable option for this project. I propose we use the MIT license.

**Comment by Esinko** [comment]
> Before this branch can be merged, we need to define what License this project uses. The utility script `discord.sh` requires us to use GPLv3 which in my opinion is not a favorable option for this project. I propose we use the MIT license.

Therefore, we need to address the need for the `discord.sh` utility script. Could @ahnl quickly summarize it's purpose?

**Comment by DrVilepis** [comment]
This is not a library so I don't exactly see why this should be licensed under MIT and not GPLv3, this wont be used in other projects anyway.

**Comment by Esinko** [comment]
> This is not a library so I don't exactly see why this should be licensed under MIT and not GPLv3, this wont be used in other projects anyway.

I guess that's fair enough. The idea of a utility script limiting our options in terms of copyright just seems foolish to me, that's all.

**Comment by ahnl** [comment]
Thank you for the observation. I agree with Esinko, especially when the library is easily replaceable. 

The utility's purpose is to deliver script's stdout to Discord web hook. We could remove it and replace it with just a curl command.

https://github.com/Testausserveri/testausapis/blob/a964e7c880830431ca9e7b2e76ae5d565558785a/scripts/syncDiscord/runWithWebhook.sh#L4

**Comment by Esinko** [comment]
> Thank you for the observation. I agree with Esinko, especially when the library is easily replaceable.
> 
> The utility's purpose is to deliver script's stdout to Discord web hook. We could remove it and replace it with just a curl command.
> 
> https://github.com/Testausserveri/testausapis/blob/a964e7c880830431ca9e7b2e76ae5d565558785a/scripts/syncDiscord/runWithWebhook.sh#L4

We already have Discord.JS as a project dependency, why not use that to send data to Discord webhooks?
https://discordjs.guide/popular-topics/webhooks.html#fetching-webhooks-of-a-channel

**Comment by Esinko** [comment]
> > Thank you for the observation. I agree with Esinko, especially when the library is easily replaceable.
> > The utility's purpose is to deliver script's stdout to Discord web hook. We could remove it and replace it with just a curl command.
> > https://github.com/Testausserveri/testausapis/blob/a964e7c880830431ca9e7b2e76ae5d565558785a/scripts/syncDiscord/runWithWebhook.sh#L4
> 
> We already have Discord.JS as a project dependency, why not use that to send data to Discord webhooks? https://discordjs.guide/popular-topics/webhooks.html#fetching-webhooks-of-a-channel

Especially when we are already running a Node program...

**Comment by ahnl** [comment]
I thought of running the webhook externally, so that we can capture also a crashed script.

File: src/database/schemas/projects.js [hunk 1]
```diff
@@ -3,10 +3,39 @@ import mongoose from "mongoose"
...
+                type: String,
+                enum : ['GITHUB', 'HOMEPAGE', 'LINK'],
+                default: 'GITHUB'
+            },
+            url: String
+        }
+    ]
 })
 
+SchemaProjects.statics.new = function (name, members) {
```
Comment (Esinko) [inline src/database/schemas/projects.js:37]: Why is this function empty?

Comment (ahnl) [reply src/database/schemas/projects.js:37]: Good catch, I forgot it there. I'll finish it once the feature will become necessary.

Comment (Esinko) [reply src/database/schemas/projects.js:37]: I'm working on making ESLint happy with this PR so I'll comment that out.

**Pull request merged**